pkg/kubelet/kubelet.go

```diff
@@ -2025,11 +2025,12 @@ func (kl *Kubelet) handleMirrorPod(mirrorPod *v1.Pod, start time.Time) {
 func (kl *Kubelet) HandlePodAdditions(pods []*v1.Pod) {
 	start := kl.clock.Now()
 	sort.Sort(sliceutils.PodsByCreationTime(pods))
+	// Responsible for checking limits in resolv.conf
+	// The limits do not have anything to do with individual pods
+	if kl.dnsConfigurer != nil && kl.dnsConfigurer.ResolverConfig != "" {
```

Comment:
I'm not sure when the check should be performed. I'm not even sure why we placed the checks in pod add/update events. @MrHohn, do you have any idea?

Reply:
Good catch.. As this has nothing to do with pod, why didn't we call it just once upon kubelet startup? Maybe this was for preventing the resolv.conf file becomes invalid after kubelet startup (e.g. file edited after kubelet runs)? It seems like we read config from this file and generate pod dns config upon pod creation:
https://github.com/kubernetes/kubernetes/blob/a44ef3079dbb188979c7de7f2d15e9c39800c869/pkg/kubelet/network/dns/dns.go#L323-L328

Reply:
Since HandlePodAdditions accepts several Pods, it seems we can afford the cost of resolv.conf validation outside the loop.

HandlePodUpdates has similar issue.

Will push an update.